### DISCUSSION TITLE
Read Environment Variables in Foundry commands

### DISCUSSION THREAD

**Discussion by alymurtazamemon** [foundry-rs/foundry#5039] [Q&A]
```bash
$ forge create --rpc-url <your_rpc_url> --private-key <your_private_key> src/MyContract.sol:MyContract
```
This is the basic command for deploying the contract. Does anyone have any idea how we can pass the <your_private_key> from the .env file instead of passing it in the command?

I read a discussion where someone suggests writing it to bash script and read from there but this is very insecure.

**Comment by boringslav** [comment]
@alymurtazamemon 
To be able to use the  environment variables  from your .env file in the shell you have to execute the content of the .env file in the current shell environment:

```bash
#Execute the content of the .env file in the current shell environment
source .env
#You can check if it is working by printing the variable in the terminal
echo $MAINNET_RPC_URL
# run your command with the variables from the .env file
forge create --rpc-url $MAINNET_RPC_URL --private-key $PRIVATE_KEY src/MyContract.sol:MyContract
```

```yaml
#This is your .env file
MAINNET_RPC_URL="rpc url"
PRIVATE_KEY="your private key"
```

**Reply by alymurtazamemon** [reply]
@boringslav Thanks, ya I found it after that!
